src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseExceptionThrowHelpersTests.cs

```diff
@@ -0,0 +1,1171 @@
+
+        if (arg is null)
+        {
+            throw new ArgumentNullException(nameof(arg));
+        }
+        else if (arg == ""test"")
+        {
+            Console.WriteLine(arg);
+        }
+
+        throw new ArgumentNullException(nameof(arg)); // no guard
+    }
+
+    string this[string name]
+    {
+        get
+        {
+            ArgumentNullException.ThrowIfNull(name);
+            return name;
+        }
+    }
+}
+"
+            }.RunAsync();
+        }
+
+
+        [Fact]
+        public async Task ArgumentNullExceptionThrowIfNull_EnsureSystemIsUsed()
+        {
+            await new VerifyCS.Test()
+            {
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net60,
+                TestCode =
+@"
+class C
+{
+    void M(string arg)
+    {
+        {|CA1510:if (arg is null) throw new System.ArgumentNullException(nameof(arg));|}
+    }
+}
+",
+                FixedCode =
+@"
+class C
+{
+    void M(string arg)
+    {
+        System.ArgumentNullException.ThrowIfNull(arg);
+    }
+}
+"
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ArgumentExceptionThrowIfNullOrEmpty_DoesntExist_NoDiagnostics()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+namespace System
+{
+    public class ArgumentException : Exception
+    {
+        public ArgumentException() { }
+        public ArgumentException(string message) { }
+        public ArgumentException(string message, string paramName) { }
+        public ArgumentException(string message, string paramName, Exception innerException) { }
+    }
+}
+
+class C
+{
+    void M(string arg)
+    {
+        if (string.IsNullOrEmpty(arg))
+            throw new ArgumentException("""", ""arg"");
+
+        if (arg is null || arg.Length == 0)
+            throw new ArgumentException("""", ""arg"");
+
+        if (arg == null || 0 == arg.Length)
+            throw new ArgumentException("""", ""arg"");
+
+        if (arg is null || arg == string.Empty)
+        {
+            throw new ArgumentException("""", ""arg"");
+        }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task ArgumentExceptionThrowIfNullOrEmpty_FixAppliesAsAppropriate()
+        {
+            await new VerifyCS.Test()
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestCode =
+@"
+using System;
+
+namespace System
+{
+    public class ArgumentException : Exception
+    {
+        public ArgumentException() { }
+        public ArgumentException(string message) { }
+        public ArgumentException(string message, string paramName) { }
+        public ArgumentException(string message, string paramName, Exception innerException) { }
+        public static void ThrowIfNullOrEmpty(string arg) { }
+    }
+}
+
+class C
+{
+    void M0(string arg)
+    {
+        {|CA1511:if (string.IsNullOrEmpty(arg))
+            throw new ArgumentException("""", ""arg"");|}
```

Comment:
Would it also flagged with a null message? If so could you change some of the `""` with `null`?